runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -43,50 +36,155 @@ void configureHeader();
 void insertNoOpCommand(unsigned int numPadBytes);
 }
 
-#define OSTREAM_OP(O_TYPE, TYPE) O_TYPE &operator<<(O_TYPE &os, const TYPE &s);
-
 namespace mlir::iree_compiler::AMDAIE {
-#define TO_STRING(TYPE) std::string to_string(const TYPE &t);
+struct TileLoc {
+  int col, row;
+  inline bool operator<(const TileLoc& rhs) const {
+    return std::tie(col, row) < std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator==(const TileLoc& rhs) const {
+    return std::tie(col, row) == std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
+
+  operator XAie_LocType() const { return XAie_TileLoc(col, row); }
+  TileLoc(XAie_LocType loc) : col(loc.Col), row(loc.Row) {}
+  TileLoc(int col, int row) : col(col), row(row) {}
```

Comment:
Can never remember C++ rule for uninitialised class int members, do you want TileLoc() : TileLoc(0,0) {} or TileLoc() = delete ?

Reply:
definitely `delete`. Thanks.